packages/@aws-cdk/aws-iot-actions/lib/cloudwatch-alarm-action.ts

```diff
@@ -0,0 +1,49 @@
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import { CommonActionProps } from './common-action-props';
+import { singletonActionRole } from './private/role';
+
+/**
+ * Configuration properties of an action for CloudWatch alarm.
+ */
+export interface CloudWatchAlarmActionProps extends CommonActionProps {
+  /**
+   * The reason for the alarm change.
+   */
+  readonly stateReason: string;
+  /**
+   * The value of the alarm state.
+   */
+  readonly stateValue: string;
+}
+
+/**
+ * The action to change a CloudWatch alarm state.
+ */
+export class CloudWatchAlarmAction implements iot.IAction {
```

Comment:
Not sure I love this name. It's missing a verb. How about `CloudWatchSetAlarmAction`?

Reply:
Probably we can use the official CloudWatch API name: `SetAlarmState`.

So the class name could be `CloudWatchSetAlarmStateAction`

Reply:
Sure. It's a little long, that's why I went with the shorter `CloudWatchSetAlarmAction`, but I don't hate `CloudWatchSetAlarmStateAction`.